### ISSUE TITLE
Performance collapse on Tool Use task after ~400 steps during reproduction

### ISSUE THREAD

**Issue by ZechuanWang** [lasgroup/SDPO#25]
Hi authors,

Thank you for the impressive work on SDPO. I am currently trying to reproduce the results on the Tool Use task using the official codebase and the default configuration provided.

Observations:

Initial Phase: In the early stages of training (0-400 steps), my reproduction results align well with the paper. The tooluse/acc/mean@16 reaches approximately 0.65 - 0.68, which matches the "1h" result (68.0) reported in Table 3 for Qwen3-8B + SDPO.
The Collapse: However, I observed a significant performance collapse starting around step 400. The accuracy drops drastically and eventually hits near zero, as shown in the attached tensorboard logs.

<img width="2160" height="653" alt="Image" src="https://github.com/user-attachments/assets/043ee75d-ac6d-4842-9708-4a89205a49b1" />

Questions:

Have you encountered similar instability or collapse issues during long-term training on the Tool Use task?
Are there specific hyperparameters (e.g., Learning Rate, KL coefficient, or Teacher EMA rate) that are critical for maintaining stability after the initial convergence?
Could you clarify if the "5h" result in Table 3 implies the model was trained stably for 5 hours, or does it refer to the best checkpoint found within a 5-hour window?
Any insights or suggestions would be greatly appreciated!

Environment:

Model: Qwen3-8B
Task: Tool Use 
Config: Default settings from the repo

**Comment by jonhue** [comment]
Hi @ZechuanWang,

Thanks for the detailed summary of your run! Depending on dataset and model, we also saw some collapse during later stages of training.
Parameters that were impactful for us were teacher EMA rate and the choice of loss (Jensen-Shannon is often much more stable than reverse/forward KL). We haven't played much with the batch size yet, but I could imagine that increasing it might help.

Let me know how it goes!

**Comment by Doow501** [comment]
> Hi authors,
> 
> Thank you for the impressive work on SDPO. I am currently trying to reproduce the results on the Tool Use task using the official codebase and the default configuration provided.
> 
> Observations:
> 
> Initial Phase: In the early stages of training (0-400 steps), my reproduction results align well with the paper. The tooluse/acc/mean@16 reaches approximately 0.65 - 0.68, which matches the "1h" result (68.0) reported in Table 3 for Qwen3-8B + SDPO. The Collapse: However, I observed a significant performance collapse starting around step 400. The accuracy drops drastically and eventually hits near zero, as shown in the attached tensorboard logs.
> 
> <img alt="Image" width="2000" height="653" src="https://private-user-images.githubusercontent.com/96179759/556737608-043ee75d-ac6d-4842-9708-4a89205a49b1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUwNDA4MTgsIm5iZiI6MTc3NTA0MDUxOCwicGF0aCI6Ii85NjE3OTc1OS81NTY3Mzc2MDgtMDQzZWU3NWQtYWM2ZC00ODQyLTk3MDgtNGE4OTIwNWE0OWIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDAxVDEwNDgzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZTU0Y2FmZTEyZmRiODg2ZDJhMmNjZDU1M2EwMDdjNzYyOTFiMGIzZmRiM2I4MTIyMGQyMDE1MjQ1ODMwZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3pEONF2M5BeIPjyUkbgChm9CLVtfj82usWJMQFmYeyE">
> Questions:
> 
> Have you encountered similar instability or collapse issues during long-term training on the Tool Use task? Are there specific hyperparameters (e.g., Learning Rate, KL coefficient, or Teacher EMA rate) that are critical for maintaining stability after the initial convergence? Could you clarify if the "5h" result in Table 3 implies the model was trained stably for 5 hours, or does it refer to the best checkpoint found within a 5-hour window? Any insights or suggestions would be greatly appreciated!
> 
> Environment:
> 
> Model: Qwen3-8B Task: Tool Use Config: Default settings from the repo

Hi, I encountered the same training collapse issue with SDPO on the Tool Use task. Did you manage to resolve it? If so, could you share what worked for you? Any advice would be greatly appreciated. Thanks!
